tests/generate_source/join_srcs.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+
+# Copyright 2018-2020 Arm Limited.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import argparse
+import os
+import sys
+
+parser = argparse.ArgumentParser(description='Join input sources into single output file.')
+parser.add_argument('--src-dir', action='store', required=True, help='Dir containing source files')
+parser.add_argument('--use-a', action='store_true', help='Use file A as input')
+parser.add_argument('--use-c', action='store_true', help='Use file C as input')
+parser.add_argument('--out', dest='output', action='store', required=True, help='Output file')
+args = parser.parse_args()
+
+inputs = []
+if args.use_a:
+    inputs.append(os.path.join(args.src_dir, 'an.implicit.src'))
+if args.use_c:
+    inputs.append(os.path.join(args.src_dir, 'cn.src'))
+
+with open(args.output, 'w') as outfile:
+    for infile in inputs:
+        if not os.path.exists(infile):
+            print("Input file doesn't exist: " + infile)
+            sys.exit(1)
+        outfile.write(open(infile, 'r').read() + '\n')
```

Comment:
Would it be more "pythonic" to `try/catch` the error?

```
try:
    with open(infile, 'r').read():
        ...
except (IOError, OSError) as e: # IIRC reading from a file can throw either of these
    print(e)
```

And it looks like you're adding the newlines here? Is there a particular reason for that? It seems like it's just compensating for the fact that your text editor hasn't added them to the source files, unless I've missed something and it's part of the test somehow?

Reply:
I add newlines to not depend on input files having them or not.

Reply:
Because it is possible to join 2 input files with this script.